src/V3TraceDecl.cpp

```diff
@@ -489,8 +489,9 @@ class TraceDeclVisitor final : public EmitCBaseVisitor {
             // Call all sub functions for this path
             for (AstCFunc* const subFuncp : item.second) {
                 AstCCall* const callp = new AstCCall{flp, subFuncp};
+                callp->dtypeSetVoid();
                 callp->argTypes("tracep");
-                addToTopFunc(callp);
+                addToTopFunc(callp->makeStmt());
```

Comment:
No, they are different things really.

- Ideally (in a future patch), dtype of a CCall should be set in the CCall constructor based on the return type of the function being called. I didn't do that in this patch because it's all a bit of a mess, with dtypes on CFuncs missing in some cases, so is a bit more involved.

- Extending from the above, you could have a non-void expression in statement position. E.g. a function or method call that returns a non-void value, but also has side-effects is reasonable to call in statement position. (One could then implement the 'ignored return value' warning by spotting a non-void call under an AstStmtMath without a cast for example, but that's just a sidenote)